frontend/mobile/app/(design-system)/components/Calendar/Calendar.tsx

```diff
@@ -1,106 +1,626 @@
+            record_translateX_array.current[
+                mainIndex === 0 ? 2 : mainIndex - 1
+            ] = width;
+
+            // Animate the views to their new positions:
+            let animation_array = translateX_array.current.map((_, index) => {
+                if (index !== (mainIndex === 0 ? 2 : mainIndex - 1)) {
+                    return Animated.timing(translateX_array.current[index], {
+                        toValue: record_translateX_array.current[index],
+                        useNativeDriver: false,
+                        duration: 150,
+                        easing: Easing.out(Easing.ease)
+                    });
+                }
+            }) as Animated.CompositeAnimation[];
+            Animated.parallel(animation_array, { stopTogether: true }).start(
+                async () => {
+                    // Update the main index:
+                    main_index.current = (mainIndex + 1) % 3;
+
+                    await handleCalendarDatePan('left', mainIndex);
+                }
+            );
+
+            return;
+        }
+        // Card full swipe right:
+        else {
+            // Set the current index translateX value to the positive width of the view:
+            record_translateX_array.current[mainIndex] = width;
+
+            // Set the previous index translateX value to 0:
+            record_translateX_array.current[
+                mainIndex === 0 ? 2 : mainIndex - 1
+            ] = 0;
+
+            // Set the next index translateX value to be the negative width of the view:
+            record_translateX_array.current[(mainIndex + 1) % 3] = -width;
+
+            // Animate the views to their new positions:
+            let animation_array = translateX_array.current.map((_, index) => {
+                if (index !== (mainIndex + 1) % 3) {
+                    return Animated.timing(translateX_array.current[index], {
+                        toValue: record_translateX_array.current[index],
+                        useNativeDriver: false,
+                        duration: 150,
+                        easing: Easing.out(Easing.ease)
+                    });
+                }
+            }) as Animated.CompositeAnimation[];
+            Animated.parallel(animation_array, { stopTogether: true }).start(
+                async () => {
+                    await handleCalendarDatePan('right', mainIndex);
+                    // Update the main index:
+                    main_index.current = mainIndex === 0 ? 2 : mainIndex - 1;
+                }
+            );
+            return;
+        }
+    };
+
+    const handleStateChanged = ({
+        nativeEvent
+    }: {
+        nativeEvent: Readonly<
+            GestureEventPayload & PanGestureHandlerEventPayload
+        >;
+    }) => {
+        // If the finger has been lifted
+        if (nativeEvent.state === State.END) {
+            old_translateX.current = 0;
+            handlePanAnimation(main_index.current);
```

Comment:
curious to know if this is where we can avoid lag on data fetch swipe. Decouple from pan animation possibly?